liota/lib/transports/mqtt.py

```diff
@@ -78,6 +79,14 @@ def on_disconnect(self, client, userdata, rc):
         self._connect_result_code = sys.maxsize
         self._disconnect_result_code = rc
         log.info("Disconnected with result code : {0} : {1} ".format(str(rc), paho.connack_string(rc)))
+        if self._disconnect_result_code == 0:
+            log.info("Clean disconnection")
+        else:
+            # Added a random sleep so that every client tries out reconnection to broker at different
+            # time to reduce load
+            wait_time = random.randint(1, 5)
+            log.info("Unexpected disconnection! Reconnecting in {0:d} seconds".format(wait_time))
+            time.sleep(wait_time)
```

Comment:
As per comments of this function, it can be called by a client disconnection as well. Why should we wait here if it is a client initiated disconnection?

Reply:
This case is handled in the "if" loop above.